run_unittests.py

```diff
@@ -1895,6 +1895,17 @@ def test_flock(self):
                     exception_raised = True
         self.assertTrue(exception_raised, 'Double locking did not raise exception.')
 
+    def test_check_module_linking(self):
+        '''
+        Test that shared modules are not linked with targets(link_with:) #2865
+        '''
+        tdir = os.path.join(self.unit_test_dir, '25 shared_mod linking')
+        out = self.init(tdir)
+        for expected in [
+            r'WARNING: Linking shared modules to targets is not recommended'
+        ]:
+            self.assertRegex(out, re.escape(expected))
```

Comment:
There is no need to use a regex here. This can be done much simpler with something like:

```python
    msg = r'WARNING: <rest of message>'
    self.assertTrue(msg in out)
```

Reply:
If you're going to do that, then: `self.assertIn(msg, out)`